### PR TITLE
Brave Today: dismiss intro card if it has been viewed for 4 seconds

### PR DESCRIPTION
**Pull request by petemill**
Stores a profile pref when the Intro Card (and it's Tab) have been in view for 4 seconds. Does not immediately get removed from page, it just won't show up on the next page load.

Resolves 

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_new_tab_ui/containers/app.tsx [hunk 1]
```diff
@@ -27,37 +27,40 @@ interface Props {
...
-          saveShowAddCard={PreferencesAPI.saveShowAddCard}
-          saveShowGemini={PreferencesAPI.saveShowGemini}
-          saveShowBitcoinDotCom={PreferencesAPI.saveShowBitcoinDotCom}
-          saveShowCryptoDotCom={PreferencesAPI.saveShowCryptoDotCom}
-          saveBrandedWallpaperOptIn={PreferencesAPI.saveBrandedWallpaperOptIn}
-        />
-      )
+  // don't render if user prefers an empty page
+  if (props.newTabData.showEmptyPage && !props.newTabData.isIncognito) {
+    return <div />
```
Comment (ryanml) [inline components/brave_new_tab_ui/containers/app.tsx:39]: nit: any reason not to just return `null` here?

Comment (petemill) [reply components/brave_new_tab_ui/containers/app.tsx:39]: not sure, I just moved this code from a class to a function (for no particular reason in the end as I didn't use). I think the div is for the blank page? But there's no styling. And in fact this pref is not real. So this code will never fire! We should probably remove that at some point then...

**Comment by petemill** [comment]
Seems like CI is ok, ping @bsclifton @ryanml

**Comment by bsclifton** [comment]
@petemill just needs rebase - but I can check out after that 😄

**Comment by petemill** [comment]
rebased

**Review by ryanml — APPROVED** [review]
changes lgtm

**Comment by petemill** [comment]
Linux had unrelated test failure `RewardsPromotionBrowserTest.ClaimViaPanel`

**Pull request merged**